src/codegen/trampoline.rs

```diff
@@ -0,0 +1,147 @@
+use std::io::{Result, Write};
+
+use env::Env;
+use analysis::bounds::{Bounds, BoundType};
+use analysis::parameter::Parameter;
+use analysis::ref_mode::RefMode;
+use analysis::rust_type::parameter_rust_type;
+use analysis::trampolines::Trampoline;
+use nameutil;
+use super::general::version_condition;
+use super::return_value::ToReturnValue;
+use super::sys::ffi_type::ffi_type;
+use super::trampoline_from_glib::TrampolineFromGlib;
+use super::trampoline_to_glib::TrampolineToGlib;
+use traits::IntoString;
+
+pub fn generate(w: &mut Write, env: &Env, analysis: &Trampoline,
+                in_trait: bool, object_name: &str) -> Result<()> {
+    try!(writeln!(w, ""));
+    let (bounds, end) = if in_trait {
+        ("<T>", "")
+    } else {
+        ("", " {")
+    };
+
+    let params_str = trampoline_parameters(env, analysis);
+    let func_str = func_string(env, analysis, None::<(&str, &str)>);
+    let ret_str = trampoline_returns(env, analysis);
+
+    try!(version_condition(w, env, analysis.version, false, 0));
+    try!(writeln!(w, "unsafe extern \"C\" fn {}{}({}, f: &Box_<{}>){}{}",
```

Comment:
I'd rather we continued to clean up the extern fns' signatures, put a `gpointer` here and cast it in the body.